### ISSUE TITLE
Start-Job -Scriptblock ignores closure

### ISSUE THREAD

**Issue by alx9r** [PowerShell/PowerShell#5393]
I suspect changing this behavior now would be a breaking change.  I'm mainly opening this because I'm wondering how `Start-Job -Scriptblock` is meant to be used in scenarios where the concrete data used in the scriptblock has be factored-out.  This is a common scenario when system-specific configuration data is separated from the more general automation procedures used to enact the configuration.

Steps to reproduce
------------------

```powershell
New-Module m {
    $v = 'm value'
    function f {
        param($sb)
        Start-Job $sb | Receive-Job -Wait
    }
} | Import-Module

$v = 'outer value'

& {
    $v = 'local value'
    f -sb { $using:v }.GetNewClosure()
}
```

Expected behavior
-----------------
Output: `local value`

Actual behavior
---------------

Output:  `m value`

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value                                                                                                                           
----                           -----                                                                                                                           
PSVersion                      6.0.0-beta.9                                                                                                                    
PSEdition                      Core                                                                                                                            
GitCommitId                    v6.0.0-beta.9                                                                                                                   
OS                             Microsoft Windows 6.3.9600                                                                                                      
Platform                       Win32NT                                                                                                                         
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                         
PSRemotingProtocolVersion      2.3                                                                                                                             
SerializationVersion           1.1.0.1                                                                                                                         
WSManStackVersion              3.0                                                                                                                             
```

**Comment by PetSerAl** [comment]
>I'm wondering how `Start-Job -Scriptblock` is meant to be used in scenarios where the concrete data used in the scriptblock has be factored-out.

You can use `-ArgumentList` parameter:
```powershell
$sb = { param($arg) $arg }
1..10 | % { Start-Job -ScriptBlock $sb -ArgumentList $_ } | Receive-Job -Wait -AutoRemoveJob
```

**Comment by Line40** [comment]
I just spent hours figuring out that the above mentioned behaviour is screwing with my code. 

This is, at least to me, clearly a BUG! The -ScriptBlock parameter behaves differently for for different commands. If I use Invoke-Command -ScriptBlock with my closure it works just fine. If I use it with Start-Job it does not. Neither the help nor any other place for powershell documentation mentions any of that.

Thank god I found this issue finally, I was beginning to think I was just too stupid to get my code to work.
